plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/PipelineExecutionService.java

```diff
@@ -62,17 +62,16 @@ public void execute(IndexRequest indexRequest, String pipelineId, ActionListener
             IngestDocument ingestDocument = new IngestDocument(index, type, id, routing, parent, timestamp, ttl, sourceAsMap);
             try {
                 pipeline.execute(ingestDocument);
-                if (ingestDocument.isSourceModified()) {
-                    indexRequest.source(ingestDocument.getSource());
-                }
-                indexRequest.index(ingestDocument.getMetadata(IngestDocument.MetaData.INDEX));
-                indexRequest.type(ingestDocument.getMetadata(IngestDocument.MetaData.TYPE));
-                indexRequest.id(ingestDocument.getMetadata(IngestDocument.MetaData.ID));
-                indexRequest.routing(ingestDocument.getMetadata(IngestDocument.MetaData.ROUTING));
-                indexRequest.parent(ingestDocument.getMetadata(IngestDocument.MetaData.PARENT));
-                indexRequest.timestamp(ingestDocument.getMetadata(IngestDocument.MetaData.TIMESTAMP));
-                indexRequest.ttl(ingestDocument.getMetadata(IngestDocument.MetaData.TTL));
-                listener.onResponse(ingestDocument);
+                Map<IngestDocument.MetaData, String> metadataMap = ingestDocument.extractMetadata();
```

Comment:
maybe add a comment here, that it is ok to overwrite the meta data properties, because if they were set then they were copied when the ingest document was created?